Spec/RealtimeClientConnection.swift

```diff
@@ -2649,6 +2649,118 @@ class RealtimeClientConnection: QuickSpec {
                         }
                     }
                 }
+                
+                // RTN15g RTN15g1
+                context("when connection (ttl + idle interval) period has passed since last activity") {
+                    let options = AblyTests.commonAppSetup()
+                    // We want this to be > than the sum of customTtlInterval and customIdleInterval
+                    options.disconnectedRetryTimeout = 5.0
+                    var client: ARTRealtime!
+                    var connectionId = ""
+                    let customTtlInterval: TimeInterval = 1.0
+                    let customIdleInterval: TimeInterval = 1.0
+                    
+                    it("uses a new connection") {
+                        client = AblyTests.newRealtime(options)
+                        client.connect()
+                        defer { client.close() }
+                        
+                        waitUntil(timeout: testTimeout) { done in
+                            client.connection.once(.connected) { _ in
+                                expect(client.connection.id).toNot(beNil())
+                                connectionId = client.connection.id!
+                                client.connectionStateTtl = customTtlInterval
+                                client.maxIdleInterval = customIdleInterval
+                                client.connection.once(.disconnected) { _ in
+                                    let disconnectedAt = Date()
+                                    expect(client.connectionStateTtl).to(equal(customTtlInterval))
+                                    expect(client.maxIdleInterval).to(equal(customIdleInterval))
+                                    client.connection.once(.connecting) { _ in
+                                        let reconnectionInterval = Date().timeIntervalSince(disconnectedAt)
+                                        expect(reconnectionInterval).to(beGreaterThan(client.connectionStateTtl + client.maxIdleInterval))
```

Comment:
Why is this the case?  I don't see any explicit request to pause / delay the reconnecting.  What has caused this delay then?

Reply:
@mattheworiordan It was this option, which tries the reconnection after 5s the client has been in a `DISCONNECTED` state https://github.com/ably/ably-ios/blob/develop/Spec/RealtimeClientConnection.swift#L2657

Reply:
But this is not according to the spec.  The client will retry to reconnect immediately after disconnect, see https://docs.ably.io/client-lib-development-guide/features/#RTN15a, not wait `disconnectedRetryTimeout`.  It only waits `disconnectedRetryTimeout` on subsequent requests.